core/base/src/TUrl.cxx

```diff
@@ -107,6 +107,7 @@ TUrl::~TUrl()
 
 void TUrl::SetUrl(const char *url, Bool_t defaultIsFile)
 {
+   delete fOptionsMap;
```

Comment:
One other possibly is to make fOptionsMap a unique_ptr ... (Note: this is just a mild suggestion).

Reply:
> One other possibly is to make fOptionsMap a unique_ptr 

This is true for many-many places in root.
But this PR is just bugfix.